stage1/init/init.go

```diff
@@ -217,10 +203,24 @@ func getArgsEnv(p *Pod, flavor, systemdVersion string, debug bool) ([]string, []
 		} else {
 			args = append(args, fmt.Sprintf("--register=false"))
 		}
+
 	default:
 		return nil, nil, fmt.Errorf("unrecognized stage1 flavor: %q", flavor)
 	}
 
+	if util.IsRunningSystemd() && systemdSupportsJournalLinking(systemdVersion) {
```

Comment:
This is confusing because on the same if, we test systemd on the host and systemd in stage1. I wonder how to make it clearer. Maybe with a comment. and renaming `systemdVersion` to `systemdStage1Version`?